lib/diagnostic.js

```diff
@@ -5,22 +5,38 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  let result = arguments[0];
+  for (let i=0; i < arguments.length; i++) {
+    if (result > arguments[i]) {
+      result = arguments[i];
+    }
+  }
+  return result;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
+  let total = arguments[0];
 
+  for (let i=1; i < arguments.length; i++) {
+    total += arguments[i];
+  }
+  return total;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
+  let array;
 
+  for (let i=0; i < length; i++) {
+    array[i] = defaultValue;
+  }
+  return array;
```

Comment:
Didn't pass the tests. Be sure to run the tests and read them. They will give you direction on what needs to be addressed.